tests/integration/test_social_media_services.py

```diff
@@ -0,0 +1,112 @@
+# Standard Stuff
+import json
+
+# Third Party Stuff
+import pytest
+import tweepy
+from unittest import mock
+
+# nexus Stuff
+from tests import utils as u
+from django.urls import reverse
+from django.utils import timezone
+from tests import factories as f
+from nexus.social_media import services
+from nexus.social_media.models import Post
+from settings.development import TWITTER_OAUTH
+from nexus.base.exceptions import WrongArguments
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+
+pytestmark = pytest.mark.django_db
+
+
+def test_update_post_object(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'twitter',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+
+    response = client.json.post(url, json.dumps(post))
+    post_instance = Post.objects.get(pk=response.data['id'])
+
+    assert post_instance.is_posted is False
+    assert post_instance.posted_time is None
+
+    services.update_post_object(post_instance)
+
+    assert post_instance.is_posted is True
+    assert post_instance.posted_time.date() == timezone.now().date()
+
+
+def test_get_twitter_api_object(client):
+    with mock.patch('nexus.social_media.services.tweepy.OAuthHandler') as mock_oauthhandler:
```

Comment:
I don't think this test was needed. Can you please let me know the need for this?

Also, please avoid nesting within context managers. If you don't use context managers here, it would make it cleaner, IMHO.

Reply:
Thanks for your suggestion! :) I'll avoid nesting context managers.

And yeah, this test wasn't needed as such. I wrote it because:
- I wanted the complete code coverage for the code I have written.
- It helps in increasing the overall code coverage of the project.
- It just verifies that required `tweepy` functions are called with appropriate arguments. That's it.

Let me know if I should refactor this test or remove it?

Reply:
It is not your responsibility to test third-party code. 

It is okay about increasing test coverage, but it is unnecessary. You can keep it since you've written it, but you can't keep writing a test for third-party function. For example, if you're using JSON module, you won't be testing its dumps or loads etc.